src/geometamaker/cli.py

```diff
@@ -16,16 +17,68 @@
 HANDLER.setFormatter(FORMATTER)
 
 
+# The recommended approach to allowing multiple ParamTypes
+# https://github.com/pallets/click/issues/1729
+class _ParamUnion(click.ParamType):
+    def __init__(self, types, report_all_errors=True):
+        """Union of click.ParamTypes.
+
+        Args:
+            types (list): List of click.ParamTypes to try to convert the value.
+            report_all_errors (bool): If True, all errors will be reported.
+                If False, only the last error will be reported.
+
+        """
+        self.types = types
+        self.report_all_errors = report_all_errors
+
+    def convert(self, value, param, ctx):
+        errors = []
+        for type in self.types:
+            try:
+                return type.convert(value, param, ctx)
+            except click.BadParameter as e:
+                errors.append(e)
+                continue
```

Comment:
Just a minor suggestion here to avoid overwriting the reference to the `type` builtin

```suggestion
        for type_ in self.types:
            try:
                return type_.convert(value, param, ctx)
            except click.BadParameter as e:
                errors.append(e)
                continue
```